Mathlib/Algebra/Polynomial/RingDivision.lean

```diff
@@ -364,6 +364,73 @@ theorem natDegree_pos_of_monic_of_not_isUnit {a : R[X]} (hu : ¬ IsUnit a) (ha :
     0 < natDegree a :=
   natDegree_pos_iff_degree_pos.mpr <| degree_pos_of_monic_of_not_isUnit hu ha
 
+open nonZeroDivisors Nat Set in
+/-- *McCoy theorem*: a polynomial `P : R[X]` is a zerodivisor if and only if there is `a : R`
+such that `a ≠ 0` and `a • P = 0`. We follow the proof given in
+https://math.stackexchange.com/questions/83121/zero-divisor-in-rx/83171#83171. -/
+theorem nmem_nonZeroDivisors_iff (P : R[X]) : P ∉ R[X]⁰ ↔ ∃ (a : R), a ≠ 0 ∧ a • P = 0 := by
```

Comment:
Can you add `P ∈ R[X]⁰ ↔ ∀ a : R, a • P = 0 → a = 0`?

Reply:
Done.